src/System.Text.Json/tests/JsonObjectTests.cs

```diff
@@ -0,0 +1,621 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static partial class JsonObjectTests
+    {
+        [Fact]
+        public static void TestDefaultConstructor()
+        {
+            var jsonObject = new JsonObject();
+            Assert.Equal(0, jsonObject.PropertyNames.Count);
+            Assert.Equal(0, jsonObject.PropertyValues.Count);
+        }
+
+        [Fact]
+        public static void TestIEnumerableKVPConstructor()
+        {
+            var jsonProperties = new List<KeyValuePair<string, JsonNode>>();
+            jsonProperties.Add(new KeyValuePair<string, JsonNode>("number", new JsonNumber(17)));
+            jsonProperties.Add(new KeyValuePair<string, JsonNode>("string", new JsonString("property value")));
+            jsonProperties.Add(new KeyValuePair<string, JsonNode>("boolean", new JsonBoolean(true)));
+
+            var jsonObject = new JsonObject(jsonProperties);
+
+            Assert.IsType<JsonNumber>(jsonObject["number"]);
+            Assert.Equal(17, (jsonObject["number"] as JsonNumber).GetInt32());
+
+            Assert.IsType<JsonString>(jsonObject["string"]);
+            Assert.Equal("property value", (jsonObject["string"] as JsonString).Value);
+
+            Assert.IsType<JsonBoolean>(jsonObject["boolean"]);
+            Assert.True((jsonObject["boolean"] as JsonBoolean).Value);
+        }
+
+        public enum ExpectedValue
+        {
+            Previous,
+            New
+        }
+
+        private static void TestDuplicates(DuplicatePropertyNameHandling duplicatePropertyNameHandling, ExpectedValue expected = default, bool valueConstructor = true)
+        {
+            string previousValue = "value1";
+            string newValue = "value1";
```

Comment:
If previousValue and newValue are the same, how do we know the test works?

Reply:
Yes, thanks, this should of course be "value2".